contracts/SpokePool.sol

```diff
@@ -39,21 +43,45 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
     // Origin token to destination token routings can be turned on or off.
     mapping(address => mapping(uint256 => bool)) public enabledDepositRoutes;
 
+    struct RelayData {
+        address sender;
+        address recipient;
+        address destinationToken;
+        uint64 realizedLpFeePct;
+        uint64 relayerFeePct;
+        uint64 depositId;
+        uint256 originChainId;
+        uint256 relayAmount;
+    }
+
+    // Each relay is associated with the hash of parameters that uniquely identify the original deposit and a relay
+    // attempt for that deposit. The relay itself is just represented as the amount filled so far. The total amount to 
+    // relay, the fees, and the agents are all parameters included in the hash key.
+    mapping(bytes32 => uint256) relays;
+
     /****************************************
      *                EVENTS                *
      ****************************************/
     event EnabledDepositRoute(address originToken, uint256 destinationChainId, bool enabled);
     event SetDepositQuoteTimeBuffer(uint64 newBuffer);
     event FundsDeposited(
-        uint256 depositId,
         uint256 destinationChainId,
         uint256 amount,
+        uint64 depositId,
         uint64 relayerFeePct,
         uint64 quoteTimestamp,
         address originToken,
         address recipient,
         address sender
     );
+    event FilledRelay(
+        bytes32 relayHash,
```

Comment:
nit: do you think this should be indexed?

Reply:
yeah